lib/mutating-array.js

```diff
@@ -1,21 +0,0 @@
-'use strict';
-
-module.exports = Ember.Mixin.create({
```

Comment:
something about this being a mixin was causing ember to choke due to weird and terribly difficult to debug `_super` issues, so I moved it into rest-model proper